### ISSUE TITLE
Distinct outputs of line-field 

### ISSUE THREAD

**Issue by RiekertKBW** [invoice-x/invoice2data#525]
Hi,

In my latest case, I want to have the option, that I'm extracting the “partnumber” variable out of a line-item.
Now there is the case, that there is sometimes a special number of our ERPs printed on the document, sometimes not.

As an example, I have this input-text:

{...}

```
1 Test1 12 pc 1,00 12,00
Description. no/one cares
2nd description-no one cares again
12-3x/yz345     #<- "partnumber" because there is not a "Your Number:" later!!

2 Test2 1 mtr 15,00 15,00
Desc. No one cares!
2nd Desc bla bla-bla
98-7abc65/423      # <- Not relevant because there is a "last_line" with "Your Number:".
Your Number: T123AB864 # <- The "partnumber"

```
{...}

So for the second item it will show the result “987abc65423\nT123AB864”.
My preferred result would be just the “T123AB864”. I can't ignore the fact that sometimes Invoices come up where the “Your Number:” is missing and the “partnumber” is then the line before the new first-line.

I can't get a regex to work, that the last_line could include the line with the “alternative” partnumber because it's that generic, like I showed you in the example input.

Would it be possible to override the “partnumber” field in the template, that there is just a single output given back without the “\n”?

**Comment by bosd** [comment]
@RiekertKBW 
I don't fully understand the problem which you are facing here.

Looks like trouble with getting your regex to work?

You could use a different capturing group for `Your number` would that solve it?
Also tehere is a posibility to put the `\n` outside your first capturing group.


For more detailed feedback. Please provide your example input file, and template.

**Comment by RiekertKBW** [comment]
> Looks like trouble with getting your regex to work?

No the RegEx is not a problem for me. It's working.
 
> You could use a different capturing group for `Your number` would that solve it? Also tehere is a posibility to put the `\n` outside your first capturing group.

Yes that a thing.. but I want to have it as most generic as possible in the Code. 
I wanted to know if there is an option, where I can just "overwrite" the group of 'Your number'. 

So if there is the case, that 'Your number' already captured something in "first_line"(for example "Test123") 

But then the same group captured something in "line" (for example "XYZ987").

What I meant was, if there was an option, that the previous captured string ("Test123") could be deleted and the new captured string("XYZ987") should be the only one which is left. 

Now the Output is "Test123\nXYZ987"
What I preffered to have would be "XYZ987".

But yea.. for now I worked around that in my Code, like that:

`
if line['Your number'] != '':
   do something...
`
`elif line['Your number fallback'] != '':
   do something else
`
`elif line['Your number fallback 2'] != '':
   do something else else
`

> For more detailed feedback. Please provide your example input file, and template.

Thats a bit difficult, because the example files include some names, phone numbers and mails which shouldn't be sent all over the Internet ;)


But I found something for me.. so I can close the Issue.

Thank you anyway @bosd
